docs/Caching.md

```diff
@@ -0,0 +1,158 @@
+# Caching
+
+Web platform apps cache data at a few different levels in order to improve UI
+responsiveness. There are two primary types of content that are cached:
+
+1. REST API responses
+2. Static assets (files)
+
+## Overview
+
+![Cache diagram](https://user-images.githubusercontent.com/1885153/27892985-d75546fe-6256-11e7-8272-66252caa5c85.png)
```

Comment:
It looks like all data goes through the service worker in the browser. I think this is a bit misleading because

a) not all browsers have a service worker
b) the service worker is only used to provide cached static assets
c) the browser uses IndexedDB to cache API responses